clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -908,10 +908,15 @@ def BinOp : CIR_Op<"binop", [Pure,
   // TODO: get more accurate than CIR_AnyType
   let results = (outs CIR_AnyType:$result);
   let arguments = (ins Arg<BinOpKind, "binop kind">:$kind,
-                       CIR_AnyType:$lhs, CIR_AnyType:$rhs);
+                       CIR_AnyType:$lhs, CIR_AnyType:$rhs,
+                       UnitAttr:$hasNoUnsignedWrap,
+                       UnitAttr:$hasNoSignedWrap);
```

Comment:
Can we instead just have a single `UnitAttr:$wrap` and infer nsw or nuw from the type? LLVM accepts both at the same time in an instruction, I'm assuming CIR won't be able to carry both of the same time given we always have sign information? Thoughts?

We also need to teach the verifier to only accept wrapping for certain binops.